packages/fxa-content-server/app/scripts/models/subscription.js

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent current subscription state, so that metrics can
+ * associate events with products and plans.
+ *
+ * Tries to read data from the URL or, failing that, the resume token.
+ */
+
+import Backbone from 'backbone';
+import Cocktail from 'cocktail';
+import ResumeTokenMixin from './mixins/resume-token';
+import UrlMixin from './mixins/url';
+import Url from '../lib/url';
+
+const SubscriptionModel = Backbone.Model.extend({
+  initialize(options) {
+    options = options || {};
+
+    if (options.planId && options.productId) {
+      this.set('planId', options.planId);
+      this.set('productId', options.productId);
+      return;
+    }
+
+    this.window = options.window || window;
+    const params = Url.searchParams(this.window.location.search);
+
+    if (params.plan_id) {
+      this.set('planId', params.plan_id);
+      this.set('productId', this.window.location.pathname.split('/').pop());
```

Comment:
This is hackish and based on our Slack conversation. `planId` is fine, but obviously I'm making a pretty big assumption that I can always just pop `productId` from the path like that. Is there some validation/matching I can do to make it more reliable? Do product ids conform to a standard format at all?

Reply:
> Do product ids conform to a standard format at all?

Or, more sensibly, are they always underneath `product/` in the path?